R/tar-terra-wrap.R

```diff
@@ -0,0 +1,153 @@
+#' if (Sys.getenv("TAR_LONG_EXAMPLES") == "true") {
+#'  targets::tar_dir({ # tar_dir() runs code from a temporary directory.
+#'    library(geotargets)
+#'    targets::tar_script({
+#'      list(
+#'        geotargets::tar_terra_rast(
+#'          terra_rast_example,
+#'          system.file("ex/elev.tif", package = "terra") |> terra::rast()
+#'        )
+#'      )
+#'    })
+#'    targets::tar_make()
+#'    x <- targets::tar_read(terra_rast_example)
+#'  })
+#'}
+tar_terra_rast_wrap <- function(name,
+                           command,
+                           pattern = NULL,
+                           filetype = "GTiff",
+                           gdal = geotargets::geotargets_option_get("gdal.raster.creation.options"),
+                           cachedir = geotargets::geotargets_option_get("cache.dir"),
+                           ...,
+                           tidy_eval = targets::tar_option_get("tidy_eval"),
+                           packages = targets::tar_option_get("packages"),
+                           library = targets::tar_option_get("library"),
+                           repository = targets::tar_option_get("repository"),
+                           iteration = targets::tar_option_get("iteration"),
+                           error = targets::tar_option_get("error"),
+                           memory = targets::tar_option_get("memory"),
+                           garbage_collection = targets::tar_option_get("garbage_collection"),
+                           deployment = targets::tar_option_get("deployment"),
+                           priority = targets::tar_option_get("priority"),
+                           resources = targets::tar_option_get("resources"),
+                           storage = targets::tar_option_get("storage"),
+                           retrieval = targets::tar_option_get("retrieval"),
+                           cue = targets::tar_option_get("cue")) {
+    filetype <- filetype %||% "GTiff"
+    gdal <- gdal %||% character(0)
+    cachedir <- cachedir %||% "geotargets_cache"
+
+    #check that filetype option is available
+    drv <- get_gdal_available_driver_list("raster")
+    filetype <- rlang::arg_match0(filetype, drv$name)
+
+    check_pkg_installed("terra")
+
+    name <- targets::tar_deparse_language(substitute(name))
+
+    envir <- targets::tar_option_get("envir")
+
+    command <- targets::tar_tidy_eval(
+        expr = as.expression(substitute(command)),
+        envir = envir,
+        tidy_eval = tidy_eval
+    )
+
+    pattern <- targets::tar_tidy_eval(
+        expr = as.expression(substitute(pattern)),
+        envir = envir,
+        tidy_eval = tidy_eval
+    )
+
+    .format_terra_rast_wrap_write <- eval(substitute(function(object, path) {
+        # TODO: provide mapping of major file types to extensions
+        extension <- switch(filetype,
+                            "GTiff" = ".tif",
+                            "GPKG" = ".gpkg",
+                            "")
```

Comment:
Is this necessary?  We don't need file extensions when writing to `_targets/`, why do we need them here?

Reply:
It is not necessary for this `wrapCache()` approach in general. 

I think I was thinking that the files in _geotargets cache folder might be something that could be accessed directly outside of R e.g. in other GIS software where having the extension would be helpful. Probably we don't want to encourage that type of use as it could change data or metadata outside the targets pipeline, so it is probably best to remove all handling of file extensions.